src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//Ported from https://github.com/adafruit/Adafruit_Python_BMP/blob/master/Adafruit_BMP/BMP085.py  
+//Formulas and code examples can also be found in the datasheet https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+
+using System;
+using System.Device.I2c;
+using System.Buffers.Binary;
+using System.Threading;
+
+namespace Iot.Device.Bmp180
+{
+    public class Bmp180:IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private readonly CommunicationProtocol _communicationProtocol;
+        private bool _initialized = false;
+        private readonly CalibrationData _calibrationData;
+        private Sampling _mode = Sampling.Standard;
+
+        private enum CommunicationProtocol
+        {
+            I2c
+        }
+
+        public Bmp180(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;            
+            _calibrationData = new CalibrationData();
+            _communicationProtocol = CommunicationProtocol.I2c;
+        }
+
+        private void Begin()
+        {            
+            //Read the coefficients table
+            _calibrationData.ReadFromDevice(this);

```

Comment:
Could you initialize this in the ctor and remove this method/`_initialized`?

Maybe create a ctor on `CalibrationData` that accepts `Bmp180` or a static factory.

Reply:
I took as an example to codify the implementation of BMP280.

https://github.com/dotnet/iot/blob/f485b50fae21a044c1ddfa3876426a278de26eaf/src/devices/Bmp280/Bmp280.cs#L41-L50